File: src/Bzn.Cloudios.Domain/Entities/Realm.cs
namespace Bzn.Cloudios.Domain.Entities;

public sealed class Realm
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; }

    public List<User> Users { get; set; } = [];
    public List<Container> Containers { get; set; } = [];
    public List<ManagedDatabaseInstance> ManagedDatabases { get; set; } = [];

    public int? MaxContainers { get; set; }
    public int? MaxDatabases { get; set; }
    public int? MaxManagedApps { get; set; }
    public long? MaxRamBytes { get; set; }
    public double? MaxCpuCores { get; set; }
}

File: src/Bzn.Cloudios.WebPlatform/src/types/realm.ts
export interface RealmListResponse {
  items: RealmItem[];
  totalCount: number;
  page: number;
  pageSize: number;
  hasNextPage: boolean;
}

export interface RealmItem {
  id: string;
  name: string;
  slug: string;
  isActive: boolean;
  createdAt: string;
  userCount: number;
  containerCount: number;
  monthlyCostBRL: number;
}

export interface RealmDetailResponse {
  id: string;
  name: string;
  isActive: boolean;
  createdAt: string;
  users: RealmUserItem[];
}

export interface RealmUserItem {
  id: string;
  email: string;
  role: string;
  isBlocked: boolean;
}

export interface CreateRealmRequest {
  name: string;
}

export interface UpdateRealmRequest {
  name: string;
  isActive: boolean;
}

export interface RealmStatsResponse {
  usersCount: number;
  containersCount: number;
  databasesCount: number;
  managedAppsCount: number;
  monthlyCostBRL: number;
  quotas: RealmQuotas;
  usage: RealmUsage;
}

export interface RealmQuotas {
  maxContainers?: number;
  maxDatabases?: number;
  maxManagedApps?: number;
  maxRamBytes?: number;
  maxCpuCores?: number;
}

export interface RealmUsage {
  containersCount: number;
  databasesCount: number;
  managedAppsCount: number;
  ramBytesUsed: number;
  cpuCoresUsed: number;
}

export interface RealmResource {
  id: string;
  name: string;
  type: 'container' | 'database';
  status: string;
  costBRL: number;
}

export interface RealmUser {
  id: string;
  email: string;
  role: string;
  isBlocked: boolean;
  createdAt: string;
}

export interface BillingHistoryItem {
  month: string;
  costBRL: number;
}

export interface RealmDetail {
  id: string;
  name: string;
  slug: string;
  isActive: boolean;
  createdAt: string;
  ownerEmail?: string;
  resources?: RealmResource[];
  users?: RealmUser[];
  billingHistory?: BillingHistoryItem[];
  quotas?: RealmQuotas;
}
